client/components/signup/SignUpForm.jsx

```diff
@@ -0,0 +1,262 @@
+    }
+    return '';
+  }
+
+  /**
+   * Method to generate a validity message for a name (first name or last name)
+   * input in any of the signup form name fields
+   * @param {String} name - name to generate a validity message for
+   * @return {String} - Message stating the validity status of the name
+   */
+  getNameValidityMessage(name = this.state.firstName) {
+    if (name.length === 0) {
+      return 'Field required';
+    }
+    if (name.length < Constant.MIN_NAME_LENGTH) {
+      return `Min name characters is ${Constant.MIN_NAME_LENGTH}`;
+    }
+    if (name.length > Constant.MAX_NAME_LENGTH) {
+      return `Max name characters is ${Constant.MAX_NAME_LENGTH}`;
+    }
+    return '';
+  }
+
+  /**
+   * Method to generate a validity message for an email from the signup form
+   * email field
+   * @param {String} email - Email strig to generate a validity message for
+   * @return {String} - Message stating the validity status of the email
+   */
+  getEmailValidityMessage(email = this.state.email) {
+    if (email.length === 0) {
+      return 'Field required';
+    }
+    if (!validator.isEmail(email)) {
+      return 'Invalid Email';
+    }
+    return '';
+  }
+
+  /**
+   * Method to check validity of password from the signup form password
+   * controls
+   * @param {String} password - password to test validity against
+   * @return {Boolean} - True if the password is valid, otherwise false
+   */
+  checkPasswordValid(password) {
+    return (password === this.state.confirmPassword &&
+      password === this.state.password &&
+      password.length > Constant.MIN_PW_LENGTH &&
+      password.length < Constant.MAX_PW_LENGTH);
+  }
+
+  /**
+   * Method to check if name in any of the name field in the signup form
+   * contains a valid name
+   * @param {String} name - Name to check if its valid
+   * @return {Boolean} - True if the name is valid, otherwise false
+   */
+  checkNameValid(name = this.state.firstName) {
+    return (name.length >= Constant.MIN_NAME_LENGTH &&
+      name.length <= Constant.MAX_NAME_LENGTH
+    );
+  }
+
+  /**
+   * Method to enable or disable the signup button based on the validity of the
+   * input fields and the state of the signup process
+   * @return {undefined} - returns undefined
+   */
+  enableSignupButton() {
+    // TODO: check state of sign up process before enabling signup button
+    return (validator.isEmail(this.state.email)
+      && this.checkPasswordValid(
+        this.state.password
+      )
+      && this.checkNameValid(
+        this.state.lastName
+      )
+      && this.checkNameValid(
+        this.state.firstName
+      )
+    );
+  }
+
+  /**
+   * Method to handle submit events on the signup form
+   * by passing them to the appropriate dispatcher
+   * @param {Oject} event - Submit event object
+   * @return{undefined} - Returns undefined
+   */
+  handleSubmit(event) {
+    event.preventDefault();
+    const userDetails = Object.assign({}, {
```

Comment:
Necessary to check user details fetched from the form and will be used when implementing connection to the server/backend